From d8164c84b43f71b3552b8c4c9367f0b322d40ba3 Mon Sep 17 00:00:00 2001
From: Yan Wang <yan.wang@linux.intel.com>
Date: Mon, 26 Mar 2018 15:02:41 +0800
Subject: [PATCH 1358/2345] ASoC: APL/CNL: Add error check for
 apl_trace_prepare().

If apl_hdac_prepare() fails in apl_trace_prepare(), output error
log for debugging.

Signed-off-by: Yan Wang <yan.wang@linux.intel.com>
---
 sound/soc/sof/hw-apl.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sound/soc/sof/hw-apl.c b/sound/soc/sof/hw-apl.c
index 0e6c153..9584be6 100644
--- a/sound/soc/sof/hw-apl.c
+++ b/sound/soc/sof/hw-apl.c
@@ -866,10 +866,15 @@ static int apl_trace_prepare(struct snd_sof_dev *sdev)
 {
 	struct snd_sof_hda_stream *stream = sdev->dtrace_stream;
 	struct snd_dma_buffer *dmab = &sdev->dmatb;
+	int ret;
 
 	stream->bufsize = sdev->dmatb.bytes;
 
-	return apl_hdac_prepare(sdev, stream, dmab, NULL);
+	ret = apl_hdac_prepare(sdev, stream, dmab, NULL);
+	if (ret < 0)
+		dev_err(sdev->dev, "error: hdac prepare failed: %x\n", ret);
+
+	return ret;
 }
 
 static int apl_trace_init(struct snd_sof_dev *sdev, u32 *stream_tag)
-- 
2.7.4

